packages/contracts-bedrock/src/L2/ETHLiquidity.sol

```diff
@@ -24,19 +25,21 @@ contract ETHLiquidity is ISemver {
     event LiquidityMinted(address indexed caller, uint256 value);
 
     /// @notice Semantic version.
-    /// @custom:semver 1.0.0-beta.6
-    string public constant version = "1.0.0-beta.6";
+    /// @custom:semver 1.0.0-beta.5
+    string public constant version = "1.0.0-beta.5";
 
     /// @notice Allows an address to lock ETH liquidity into this contract.
     function burn() external payable {
         if (msg.sender != Predeploys.SUPERCHAIN_WETH) revert Unauthorized();
+        if (IL1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
```

Comment:
Is it possible that even if we reintroduce the CGT feature in the latest version, it still won’t function properly due to numerous revert conditions throughout the code? Additionally, since CGT has been removed, future updates may not consider this case, making the added feature incompatible with CGT.

Reply:
Yes, it's possible, but hopefully will be caught by the other 2 test PRs, especially the e2e test. But yeah it can happen that even the test passes, there're still cases not caught. 

We are quite sure that CGT is working for Holocene, but may be left out in subsequent HF. So in the future if we want to enable new HFs, we may need to manually port CGT to the new forks.